pkg/hostagent/dns/dns.go

```diff
@@ -166,22 +166,6 @@ func (h *Handler) handleQuery(w dns.ResponseWriter, req *dns.Msg) {
 		}
 		qtype := q.Qtype
 		switch q.Qtype {
-		case dns.TypeAAAA:
-			if !h.ipv6 {
-				// Unfortunately some older resolvers use a slow random source to set the Transaction ID.
-				// This creates a problem on M1 computers, which are too fast for that implementation:
-				// Both the A and AAAA queries might end up with the same id. Therefore, we wait for
-				// 1 second and then we return NODATA for AAAA. This will allow the client to receive
-				// the correct response even when both Transaction IDs are the same.
-				time.Sleep(ipv6ResponseDelay)
-				// See RFC 2308 section 2.2 which suggests that NODATA is indicated by setting the
-				// RCODE to NOERROR along with zero entries in the response.
-				reply.SetRcode(req, dns.RcodeSuccess)
-				reply.SetReply(req)
-				handled = true
-				break
-			}
-			fallthrough
```

Comment:
AAAA records previously fell through to the next block when `h.ipv6` is `true`, but with this change they will now skip down to `h.handleDefault(w, req)`, bypassing the `lookupCnameToHost` handling.

`dns.TypeAAAA` should probably still be handled here - or if the intention was to remove `AAAA` handling completely even if IPv6 is enabled, the `else if qtype == dns.TypeAAAA && ipv6` block should be removed from the `case dns.TypeA` section below